check_style.py

```diff
@@ -0,0 +1,72 @@
+"""
+Script to be run before a commit. Ensures that the code adheres to best
+style  practices by running the black formatter, pylint linter, and the
+isort import sorter.
+
+To automatically run this file on `git commit`, remove the `.sample`
+extension from the `.git/hooks/pre-commit.sample` file and replace the
+content with:
+
+```shell
+#!/bin/sh
+echo "[pre-commit] Running pre-commit hook..."
+python check_style.py
+if [ $? -ne 0 ]; then
+    echo "[pre-commit] FAIL"
+    exit 1
+fi
+echo "[pre-commit] OK"
+exit 0
+```
+If you're on Windows, replace the shebang (line 1) with:
+```
+#!C:/Progra~1/Git/usr/bin/sh.exe
+```
```

Comment:
The only change I would suggest to make this "windows first is to change the script to be windows by default and include the bash version as an alternative.

```suggestion
```shell
#!C:/Progra~1/Git/usr/bin/sh.exe
echo "[pre-commit] Running pre-commit hook..."
python check_style.py
if [ $? -ne 0 ]; then
    echo "[pre-commit] FAIL"
    exit 1
fi
echo "[pre-commit] OK"
exit 0
```
If you're not on Windows, replace the shebang (line 1) with:
```
#!/bin/sh
```
```